org.eclipse.jdt.launching/launching/org/eclipse/jdt/internal/launching/environments/ExecutionEnvironmentAnalyzer.java

```diff
@@ -104,6 +110,7 @@ public class ExecutionEnvironmentAnalyzer implements IExecutionEnvironmentAnalyz
 		mappings.put(JavaSE_18, new String[] { JavaSE_17 });
```

Comment:
mappings.put(JavaSE_10_Plus, new String[] { JavaSE_20 }); needs to be updated to 	mappings.put(JavaSE_10_Plus, new String[] { JavaSE_21 });